File: app/src/main/java/meugeninua/appfromscratch/ui/activities/base/BaseBindingActivity.java
package meugeninua.appfromscratch.ui.activities.base;

import javax.inject.Inject;

import meugeninua.appfromscratch.ui.activities.base.fragments.base.binding.Binding;

public abstract class BaseBindingActivity<B extends Binding> extends BaseActivity {

    @Inject protected B binding;

    @Override
    public void onContentChanged() {
        super.onContentChanged();
        binding.attachView(getWindow().getDecorView());
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        binding.detachView();
    }
}

File: app/src/main/java/meugeninua/appfromscratch/ui/activities/base/BaseBindingActivity.kt
package meugeninua.appfromscratch.ui.activities.base

import javax.inject.Inject

import meugeninua.appfromscratch.ui.activities.base.fragments.base.binding.Binding

abstract class BaseBindingActivity<B : Binding> : BaseActivity() {

    @Inject
    lateinit var binding: B

    override fun onContentChanged() {
        super.onContentChanged()
        binding.attachView(window.decorView)
    }

    override fun onDestroy() {
        super.onDestroy()
        binding.detachView()
    }
}
